orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/AbstractClusterWideClouddriverTask.groovy

```diff
@@ -198,12 +198,12 @@ abstract class AbstractClusterWideClouddriverTask extends AbstractCloudProviderA
       switch (location.type) {
         case Location.Type.ZONE:
           deployedServerGroups.addAll(clusterServerGroups.findAll {
-            it.zones.contains(location.value) && dsgs[it.region].contains(it.name)
+            it.zones.contains(location.value) && dsgs[it.region]?.contains(it.name)
```

Comment:
This handles a situation where you have multiple deploy stages in your pipeline.

Specifically a Deploy Stage #1 targeting us-east-1 and us-west-1 and a Deploy Stage #2 targeting us-west-1.

Previously a deploy stage was not looking up the pipeline hierarchy to determine what server groups were deployed and would have never ran into this situation ... rather it only looked at it's immediate parent which would have always had a `dsg`.